### PR TITLE
ticdc: add a new chapter in ticdc overview to show how we deal with data changes

### PR DESCRIPTION
**Pull request by hongyunyan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-overview.md [hunk 1]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
 
 - 在使用 TiCDC 实现容灾的场景下，为实现最终一致性，需要配置 [redo log](/ticdc/ticdc-sink-to-mysql.md#灾难场景的最终一致性复制) 并确保 redo log 写入的存储系统在上游发生灾难时可以正常读取。
 
+## TiCDC 如何翻译数据变更
+
+> 本章节我们只讨论 TiCDC 如何翻译上游 DML 产生的数据变更，不讨论如何处理上游 DDL 产生的变更。对于 DDL 操作，TiCDC 会获取到完整的 DDL SQL 语句，根据下游属性，转换成对应的格式发送给下游。
```
Comment (flowbehappy) [inline ticdc/ticdc-overview.md:83]:
```suggestion
> 本章节我们只讨论 TiCDC 如何翻译上游 DML 产生的数据变更，不讨论如何处理上游 DDL 产生的变更。对于 DDL 操作，TiCDC 会获取到完整的 DDL SQL 语句，根据下游的 Sink 类型，转换成对应的格式发送给下游。
```

File: ticdc/ticdc-overview.md [hunk 2]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
 
 - 在使用 TiCDC 实现容灾的场景下，为实现最终一致性，需要配置 [redo log](/ticdc/ticdc-sink-to-mysql.md#灾难场景的最终一致性复制) 并确保 redo log 写入的存储系统在上游发生灾难时可以正常读取。
 
+## TiCDC 如何翻译数据变更
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:81]: 这个小节的标题叫“实现原理”是否可以？其它一些文档里常用这个小节标题，通常是具体讲述如何实现的。

File: ticdc/ticdc-overview.md [hunk 3]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+> 本章节我们只讨论 TiCDC 如何翻译上游 DML 产生的数据变更，不讨论如何处理上游 DDL 产生的变更。对于 DDL 操作，TiCDC 会获取到完整的 DDL SQL 语句，根据下游属性，转换成对应的格式发送给下游。
+
+> 注意: TiCDC 翻译数据变更的逻辑可能会在后续版本发生调整。
+
+不同于 Mysql binlog 直接记录了上游执行的所有 DML 的 SQL 语句，TiCDC 通过实时监听上游 TiKV 各个 Region Raft Log 信息，我们会根据每个事务前后数据的差异生成对应多条的数据变更信息。TiCDC 只保证输出的变更事件和上游 TiDB 的变更是等价的，但是不保证能准确还原上游 TiDB 引起数据变更的 SQL。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:87]: 新增的内容里，有多处都用了第一人称“我们”，需要调整下表述。在官方用户文档里，除了 FAQ 文档描述问题的场景，其它地方通常不采用第一人称。博客文章里可以用第一人称。

File: ticdc/ticdc-overview.md [hunk 4]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
 
+## TiCDC 翻译数据变更的实现原理
+
+> 本章节只讨论 TiCDC 如何翻译上游 DML 产生的数据变更，不讨论如何处理上游 DDL 产生的变更。对于 DDL 操作，TiCDC 会获取到完整的 DDL SQL 语句，根据下游的 Sink 类型，转换成对应的格式发送给下游。
+
+> 注意: TiCDC 翻译数据变更的逻辑可能会在后续版本发生调整。
+
+不同于 Mysql binlog 直接记录了上游执行的所有 DML 的 SQL 语句，TiCDC 通过实时监听上游 TiKV 各个 Region Raft Log 信息，并且会根据每个事务前后数据的差异生成对应多条的数据变更信息。TiCDC 只保证输出的变更事件和上游 TiDB 的变更是等价的，但是不保证能准确还原上游 TiDB 引起数据变更的 SQL。
+
+数据变更信息会包含数据变更类型，以及变更前后的数值。事务前后数据的差异一共可能产生三种事件：
```
Comment (flowbehappy) [inline ticdc/ticdc-overview.md:89]: 还有可能会翻译成 replace into

File: ticdc/ticdc-overview.md [hunk 5]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
 
 - 在使用 TiCDC 实现容灾的场景下，为实现最终一致性，需要配置 [redo log](/ticdc/ticdc-sink-to-mysql.md#灾难场景的最终一致性复制) 并确保 redo log 写入的存储系统在上游发生灾难时可以正常读取。
 
+## TiCDC 翻译数据变更的实现原理
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:81]:
```suggestion
## TiCDC 处理数据变更的实现原理
```

File: ticdc/ticdc-overview.md [hunk 6]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+> 本章节只讨论 TiCDC 如何翻译上游 DML 产生的数据变更，不讨论如何处理上游 DDL 产生的变更。对于 DDL 操作，TiCDC 会获取到完整的 DDL SQL 语句，根据下游的 Sink 类型，转换成对应的格式发送给下游。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:83]:
```suggestion
本小节主要描述 TiCDC 如何处理上游 DML 产生的数据变更。对于上游 DDL 产生的数据变更，TiCDC 会获取到完整的 DDL SQL 语句，根据下游的 Sink 类型，转换成对应的格式发送给下游，本小节不再赘述。
```

File: ticdc/ticdc-overview.md [hunk 7]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+> 注意: TiCDC 翻译数据变更的逻辑可能会在后续版本发生调整。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:85]:
```suggestion
> **注意：**
>
> TiCDC 处理数据变更的逻辑可能会在后续版本发生调整。
```

File: ticdc/ticdc-overview.md [hunk 8]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+不同于 Mysql binlog 直接记录了上游执行的所有 DML 的 SQL 语句，TiCDC 通过实时监听上游 TiKV 各个 Region Raft Log 信息，并且会根据每个事务前后数据的差异生成对应多条的数据变更信息。TiCDC 只保证输出的变更事件和上游 TiDB 的变更是等价的，但是不保证能准确还原上游 TiDB 引起数据变更的 SQL。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:87]: 这样呢？
```suggestion
MySQL binlog 直接记录了上游执行的所有 DML 操作的 SQL 语句。与 MySQL 不同，TiCDC 则实时监听上游 TiKV 各个 Region Raft Log 的信息，并根据每个事务前后数据的差异生成对应多条 SQL 语句的数据变更信息。TiCDC 只保证输出的变更事件和上游 TiDB 的变更是等价的，不保证能准确还原上游 TiDB 引起数据变更的 SQL 语句。
```

File: ticdc/ticdc-overview.md [hunk 9]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+数据变更信息会包含数据变更类型，以及变更前后的数值。事务前后数据的差异一共可能产生三种事件：
+
+1. Delete 事件：对应会收到一条 Delete 类型的数据变更信息，包含变更前的数值。
+
+2. Insert 事件：对应会收到一条 Put 类型的数据变更信息，包含变更后的数值。
+
+3. Update 事件：对应会收到一条 Put 类型的数据变更信息，包含变更前与变更后的数值。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:95]:
```suggestion
1. `DELETE` 事件：对应会收到一条 `DELETE` 类型的数据变更信息，包含变更前的数值。

2. `INSERT` 事件：对应会收到一条 `PUT` 类型的数据变更信息，包含变更后的数值。

3. `UPDATE` 事件：对应会收到一条 `PUT` 类型的数据变更信息，包含变更前与变更后的数值。
```

File: ticdc/ticdc-overview.md [hunk 10]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+TiCDC 内部根据收到的这些数据变更信息，适配各个类型的下游来生成合适格式的数据传输给下游，如 Canal-Json， Avro等格式写入 Kafka 中，又或者是重新翻译成 SQL 语句发送给下游的 Mysql 或者 TiDB等。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:97]:
```suggestion
TiCDC 会根据收到的这些数据变更信息，适配各个类型的下游来生成合适格式的数据传输给下游。例如，生成 Canal-Json，Avro 等格式的数据写入 Kafka 中，或者重新转换成 SQL 语句发送给下游的 MySQL 或者 TiDB。
```

File: ticdc/ticdc-overview.md [hunk 11]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+目前 TiCDC 将数据变更信息适配对应的协议时，对于特定的 Update 事件，可能会其拆成 一条 Insert 事件和一条 Delete 事件, 具体可以参考文档 [将 Update 事件拆分为 Delete 和 Insert 事件](/ticdc/ticdc-behavior-change)。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:98]:
```suggestion
目前 TiCDC 将数据变更信息适配对应的协议时，对于特定的 `UPDATE` 事件，可能会将其拆成一条 `INSERT` 事件和一条 `DELETE` 事件。详见[将 Update 事件拆分为 Delete 和 Insert 事件](/ticdc/ticdc-behavior-change.md#将-update-事件拆分为-delete-和-insert-事件)。
```

File: ticdc/ticdc-overview.md [hunk 12]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+而当下游是 Mysql 或者 TiDB 时，因为 TiCDC 并非直接获取原生上游执行的 DML 语句，而是重新根据数据变更信息来生成 SQL 语句，因此不能保证写入下游的 SQL 语句和上游执行的 SQL 语句完全相同，但会保证最终结果的一致性。
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:100]:
```suggestion
当下游是 MySQL 或者 TiDB 时，因为 TiCDC 并非直接获取原生上游执行的 DML 语句，而是重新根据数据变更信息来生成 SQL 语句，因此不能保证写入下游的 SQL 语句和上游执行的 SQL 语句完全相同，但会保证最终结果的一致性。
```

File: ticdc/ticdc-overview.md [hunk 13]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+例如上游执行了以下 SQL，
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:102]:
```suggestion
例如上游执行了以下 SQL 语句：
```

File: ticdc/ticdc-overview.md [hunk 14]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+```
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:104]:
```suggestion
```sql
```

File: ticdc/ticdc-overview.md [hunk 15]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+BEGIN;
+Insert Into t1 values(1,2);
+Insert Into t1 values(2,2);
+Insert Into t1 values(3,3);
+Commit;
+
+Update t1 set b = 4 where b = 2; 
+```
+
+TiCDC 根据数据变更信息后重新生成 SQL，会向下游写以下两条 SQL:
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:116]:
```suggestion
TiCDC 将根据数据变更信息重新生成 SQL 语句，向下游写以下两条 SQL 语句：
```

File: ticdc/ticdc-overview.md [hunk 16]
```diff
@@ -78,6 +78,56 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+```
```
Comment (lilin90) [inline ticdc/ticdc-overview.md:118]:
```suggestion
```sql
```

File: ticdc/ticdc-overview.md (hunk 4 above)
Comment (hongyunyan) [reply ticdc/ticdc-overview.md:91]: 上游发生 replace into 的事件的话，我们应该也是收到 delete + update 事件。replace into 应该不算一个跟 delete / insert / update 同类层次的事件吧

Comment (flowbehappy) [reply ticdc/ticdc-overview.md:91]: https://docs.pingcap.com/tidb/stable/ticdc-faq#why-do-insertupdate-statements-from-the-upstream-become-replace-into-after-being-replicated-to-the-downstream-if-i-set-safe-mode-to-true-when-i-create-a-ticdc-replication-task
@hongyunyan

**Review by flowbehappy — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

File: ticdc/ticdc-overview.md [hunk 17]
```diff
@@ -78,6 +78,59 @@ TiCDC 作为 TiDB 的增量数据同步工具，通过 PD 内部的 etcd 实现
...
+
+数据变更信息会包含数据变更类型，以及变更前后的数值。事务前后数据的差异一共可能产生三种事件：
+
+1. `DELETE` 事件：对应会收到一条 `DELETE` 类型的数据变更信息，包含变更前的数据。
+
+2. `INSERT` 事件：对应会收到一条 `PUT` 类型的数据变更信息，包含变更后的数据。
+
+3. `UPDATE` 事件：对应会收到一条 `PUT` 类型的数据变更信息，包含变更前与变更后的数据。
+
+TiCDC 会根据收到的这些数据变更信息，适配各个类型的下游来生成合适格式的数据传输给下游。例如，生成 Canal-Json，Avro 等格式的数据写入 Kafka 中，或者重新转换成 SQL 语句发送给下游的 MySQL 或者 TiDB。
```
Comment (hfxsd) [inline ticdc/ticdc-overview.md:99]:
```suggestion
TiCDC 会根据收到的这些数据变更信息，适配各个类型的下游来生成合适格式的数据传输给下游。例如，生成 Canal-JSON、Avro 等格式的数据写入 Kafka 中，或者重新转换成 SQL 语句发送给下游的 MySQL 或者 TiDB。
```

**Review by hfxsd — APPROVED** [review]

**Review by lilin90 — COMMENTED** [review]
/approve

**Pull request merged**